Code/Tools/AssetProcessor/native/ui/MainWindow.cpp

```diff
@@ -750,12 +755,14 @@ void MainWindow::AddPatternRow(AZStd::string_view name, AssetBuilderSDK::AssetBu
     ui->sharedCacheTable->setItem(row, aznumeric_cast<int>(PatternColumns::Pattern), patternWidgetItem);
 
     // Remove button
-    auto* button = new QToolButton();
-    button->setIcon(QIcon(":/PropertyEditor/Resources/trash-small.png"));
-    AzQtComponents::PushButton::applySmallIconStyle(button);
+    auto* button = new QPushButton();
```

Comment:
Do these QWidgets (checkbox, name, type combo, pattern, and remove button) need a parent so that the memory can be freed when they are not used? I found `QTableWidget` takes ownership of `QTableWidgetItem`s, but I am not sure about the others: QPushButton, QComboBox, and QCheckBox each have an optional parent pointer argument in their constructors.

Reply:
https://doc.qt.io/qt-6/qtablewidget.html 

> Sets the given widget to be displayed in the cell in the given row and column, passing the ownership of the widget to the table.

I think this is a safe operation.